Applications/Spire/Source/Ui/OverlayPanel.cpp

```diff
@@ -43,11 +44,15 @@ namespace {
 }
 
 OverlayPanel::OverlayPanel(QWidget* body, QWidget* parent)
-    : QWidget(parent, Qt::Popup | Qt::FramelessWindowHint |
-        Qt::NoDropShadowWindowHint),
+    : QWidget(parent,
+        Qt::Tool | Qt::FramelessWindowHint | Qt::NoDropShadowWindowHint),
       m_body(body),
       m_is_closed_on_blur(true),
-      m_positioning(Positioning::PARENT) {
+      m_is_draggable(true),
+      m_was_activated(false),
+      m_positioning(Positioning::PARENT),
+      m_focus_observer(*this),
+      m_parent_focus_observer(*parent) {
```

Comment:
Don't dereference a value without a guarantee about its correctness.

Reply:
In that case, I think OverlayPanel, DropDownList, and FilterPanel should take a parent by reference in their constructors instead of as a pointer.

Reply:
If they are all required to have a parent, then absolutely. Good call on that.